### DISCUSSION TITLE
Create a new hook for write

### DISCUSSION THREAD

**Discussion by StKostyantin** [wevm/wagmi#1430] [Ideas]
At the moment, to make the wtite method, you need to write a lot of the same type of code.

For example:
```
const { config: config, error: error } = usePrepareContractWrite({
    address,
    abi,
    functionName,
    args,
    enabled,
  });

  const { data: dataWrite, write, isLoading: isLoadingWrite } = useContractWrite(config);

  const {
    data: dataWait,
    isLoading: isLoading,
    isSuccess: isSuccess,
    isError: isError,
  } = useWaitForTransaction({
    hash: dataWrite?.hash,
  });
```
You need to do this every time you want to do some write method, and if you have 2-3 (or more) approve and then some other methods, then it turns into a little hell. 

My idea is to create one more hook that will accept parameters for usePrepareContractWrite, and return the state of loads, errors and the write method itself.

For example:
```
const { config, errorPrepare, loadingPrepare, dataWrite, write, isLoadingWrite, dataWait, ...etc. } = useSomeHookName({
    address,
    abi,
    functionName,
    args,
    enabled,
  });
```

**Comment by tmm** [comment]
Interesting idea. wagmi is a lower-level framework so unlikely we would add this to the core at the moment. You could write a CLI plugin for this though: https://wagmi.sh/cli/plugins
